Source/Core/Core/IOS/Network/KD/Mail/WC24Receive.cpp

```diff
@@ -0,0 +1,282 @@
+
+#include "Core/IOS/Network/KD/Mail/WC24Receive.h"
+#include "Common/Assert.h"
+#include "Core/IOS/FS/FileSystem.h"
+#include "Core/IOS/Uids.h"
+
+namespace IOS::HLE::NWC24::Mail
+{
+constexpr const char RECEIVE_LIST_PATH[] = "/" WII_WC24CONF_DIR "/mbox"
+                                           "/wc24recv.ctl";
+
+WC24ReceiveList::WC24ReceiveList(std::shared_ptr<FS::FileSystem> fs) : m_fs{std::move(fs)}
+{
+  if (!ReadReceiveList())
+  {
+    ERROR_LOG_FMT(IOS_WC24, "There is an error in the Receive List for WC24 mail. Mail will be "
+                            "unavailable for this IOS session.");
+    m_is_disabled = true;
+
+    // If the Send list is corrupted, delete it.
+    const FS::ResultCode result = m_fs->Delete(PID_KD, PID_KD, RECEIVE_LIST_PATH);
+    if (result != FS::ResultCode::Success && result != FS::ResultCode::NotFound)
+    {
+      ERROR_LOG_FMT(IOS_WC24, "Failed to delete the Receive list.");
+    }
+  }
+}
+
+bool WC24ReceiveList::ReadReceiveList()
+{
+  const auto file = m_fs->OpenFile(PID_KD, PID_KD, RECEIVE_LIST_PATH, FS::Mode::Read);
+  if (!file || !file->Read(&m_data, 1))
+  {
+    ERROR_LOG_FMT(IOS_WC24, "Failed to read the Receive list");
+    return false;
+  }
+
+  if (!CheckReceiveList())
+  {
+    ERROR_LOG_FMT(IOS_WC24, "There is an error in the Receive List for WC24 mail");
+    return false;
+  }
+
+  return true;
+}
+
+void WC24ReceiveList::WriteReceiveList() const
+{
+  ASSERT(!IsDisabled());
+  constexpr FS::Modes public_modes{FS::Mode::ReadWrite, FS::Mode::ReadWrite, FS::Mode::ReadWrite};
+  m_fs->CreateFullPath(PID_KD, PID_KD, RECEIVE_LIST_PATH, 0, public_modes);
+  const auto file = m_fs->CreateAndOpenFile(PID_KD, PID_KD, RECEIVE_LIST_PATH, public_modes);
+
+  if (!file || !file->Write(&m_data, 1))
+    ERROR_LOG_FMT(IOS_WC24, "Failed to open or write WC24 Receive list file");
+}
+
+bool WC24ReceiveList::CheckReceiveList() const
+{
+  // 'WcTF' magic
+  if (Common::swap32(m_data.header.magic) != MAIL_LIST_MAGIC)
+  {
+    ERROR_LOG_FMT(IOS_WC24, "Receive List magic mismatch ({} != {})",
+                  Common::swap32(m_data.header.magic), MAIL_LIST_MAGIC);
+    return false;
+  }
+
+  if (Common::swap32(m_data.header.version) != 4)
+  {
+    ERROR_LOG_FMT(IOS_WC24, "Receive List version mismatch");
+    return false;
+  }
+
+  return true;
+}
+
+bool WC24ReceiveList::IsDisabled() const
+{
+  return m_is_disabled;
+}
+
+u32 WC24ReceiveList::GetNextEntryId() const
+{
+  ASSERT(!IsDisabled());
+  return Common::swap32(m_data.header.next_entry_id);
+}
+
+u32 WC24ReceiveList::GetNextEntryIndex() const
+{
+  ASSERT(!IsDisabled());
+  return (Common::swap32(m_data.header.next_entry_offset) - 128) / 128;
+}
+
+void WC24ReceiveList::InitFlag(u32 index)
+{
+  ASSERT(!IsDisabled());
+  m_data.entries[index].flag = 0x200200;
+}
+
+void WC24ReceiveList::FinalizeEntry(u32 index)
+{
+  ASSERT(!IsDisabled());
+  u32 next_entry_index = std::numeric_limits<u32>::max();
```

Comment:
```suggestion
  u32 next_entry_index = UINT32_MAX;
```

is a bit more concise here